packages/core/src/tools/tool-compatibility/index.ts

```diff
@@ -0,0 +1,394 @@
+              break;
+            }
+            case 'max': {
+              currentConstraints.maxLength = check.value;
+              break;
+            }
+          }
+        } else {
+          // If we're not handling this check specially, preserve it
+          newChecks.push(check);
+        }
+      }
+    }
+
+    if (Object.keys(currentConstraints).length > 0) {
+      constraints[path] = {
+        ...constraints[path],
+        ...currentConstraints,
+      };
+    }
+
+    // Return a string type with any remaining checks
+    let result = z.string();
+    for (const check of newChecks) {
+      result = result._addCheck(check);
+    }
+    return result as ShapeValue<T>;
+  }
+
+  public defaultZodNumberHandler<T extends z.AnyZodObject>(
+    value: z.ZodTypeAny,
+    path: string,
+    constraints: SchemaConstraints,
+    handleChecks: readonly NumberCheckType[] = ALL_NUMBER_CHECKS, // Default to handling all checks
+  ): ShapeValue<T> {
+    const zodNumber = value as z.ZodNumber;
+    const currentConstraints: SchemaConstraints[string] = {};
+    const checks = zodNumber._def.checks || [];
+    type ZodNumberCheck = (typeof checks)[number];
+    const newChecks: ZodNumberCheck[] = [];
+
+    for (const check of checks) {
+      if ('kind' in check) {
+        if (handleChecks.includes(check.kind as NumberCheckType)) {
+          switch (check.kind) {
+            case 'min':
+              if (check.inclusive) {
+                currentConstraints.gte = check.value;
+              } else {
+                currentConstraints.gt = check.value;
+              }
+              break;
+            case 'max':
+              if (check.inclusive) {
+                currentConstraints.lte = check.value;
+              } else {
+                currentConstraints.lt = check.value;
+              }
+              break;
+            case 'multipleOf': {
+              currentConstraints.multipleOf = check.value;
+              break;
+            }
+          }
+        } else {
+          // If we're not handling this check specially, preserve it
+          newChecks.push(check);
+        }
+      }
+    }
+
+    if (Object.keys(currentConstraints).length > 0) {
+      constraints[path] = {
+        ...constraints[path],
+        ...currentConstraints,
+      };
+    }
+
+    // Return a number type with any remaining checks
+    let result = z.number();
+    for (const check of newChecks) {
+      switch (check.kind) {
+        case 'int':
+          result = result.int();
+          break;
+        case 'finite':
+          result = result.finite();
+          break;
+        default:
+          result = result._addCheck(check);
+      }
+    }
+    return result as ShapeValue<T>;
+  }
+
+  public process(tool: ToolToConvert): {
+    description?: string;
+    parameters: Schema;
+  } {
+    if (isVercelTool(tool)) {
+      return {
+        description: tool.description,
+        // TODO: should we also process vercel tool params?
```

Comment:
@DanielSLew what do you think?

Reply:
I wasn't able to get vercel tools to work, there's another ticket to fix vercel tools, so I'd say let's ship this as is, and fix that after